src/ILCompiler.WebAssembly/src/CodeGen/EvaluationStack.cs

```diff
@@ -0,0 +1,524 @@
+        /// <returns>String representation of current entry</returns>
+        public override sealed string ToString()
+        {
+            StringBuilder s = new StringBuilder();
+            BuildRepresentation(s);
+            return s.ToString();
+        }
+
+        /// <summary>
+        /// Build a representation of current entry in <paramref name="s"/>.
+        /// </summary>
+        /// <param name="s">StringBuilder where representation will be saved.</param>
+        protected virtual void BuildRepresentation(StringBuilder s)
+        {
+            Debug.Assert(s != null, "StringBuilder is null.");
+            if (Type != null)
+            {
+                s.Append(Type);
+                if (Kind != StackValueKind.Unknown)
+                {
+                    s.Append('(');
+                    s.Append(Kind);
+                    s.Append(')');
+                }
+            }
+            else if (Kind != StackValueKind.Unknown)
+            {
+                if (Kind != StackValueKind.Unknown)
+                {
+                    s.Append('(');
+                    s.Append(Kind);
+                    s.Append(')');
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Abstract entry for all constant values.
+    /// </summary>
+    internal abstract class ConstantEntry : StackEntry
+    {
+        protected ConstantEntry(StackValueKind kind, LLVMValueRef llvmValue, TypeDesc type = null) : base(kind, llvmValue, type)
+        {
+        }
+
+        /// <summary>
+        /// Does current entry require a cast to be assigned to <paramref name="destType"/>?
+        /// </summary>
+        /// <param name="destType">Type of destination</param>
+        /// <returns>True if a cast is required</returns>
+        public virtual bool IsCastNecessary(TypeDesc destType)
+        {
+            return false;
+        }
+    }
+
+    internal abstract class ConstantEntry<T> : ConstantEntry where T : IConvertible
+    {
+        public T Value { get; }
+
+        protected ConstantEntry(StackValueKind kind, T value, LLVMValueRef llvmValue, TypeDesc type = null) : base(kind, llvmValue, type)
+        {
+            Value = value;
+        }
+
+        /*public override void Append(CppGenerationBuffer _builder)
+        {
+            _builder.Append(Value.ToStringInvariant());
+        }*/
+
+        protected override void BuildRepresentation(StringBuilder s)
+        {
+            base.BuildRepresentation(s);
+            if (s.Length > 0)
+            {
+                s.Append(' ');
+            }
+            s.Append(Value);
+        }
+    }
+
+    internal class Int32ConstantEntry : ConstantEntry<int>
+    {
+        public Int32ConstantEntry(int value, TypeDesc type = null) : base(StackValueKind.Int32, value, LLVM.ConstInt(LLVM.Int32Type(), (ulong)value, LLVMMisc.False), type)
+        {
+        }
+
+        /*public override void Append(CppGenerationBuffer _builder)
```

Comment:
Unused?

Reply:
Deleted all of the Append methods from this file.